doc/services/shell/index.rst

```diff
@@ -714,20 +716,20 @@ An example thread safe usage:
 .. code-block:: c
 
   char *cvalue = NULL;
-  struct getopt_state *state;
-  while ((char c = getopt(argc, argv, "abhc:")) != -1) {
-        state = getopt_state_get();
+  struct getopt_data state = GETOPT_DATA_INITIALIZER;
+  while ((char c = getopt_r(argc, argv, "abhc:", &state)) != -1) {
         switch (c) {
         case 'c':
-                cvalue = state->optarg;
+                cvalue = state.optarg;
                 break;
         default:
                 break;
         }
   }
 
-Thread safe getopt functionality is activated by
-:kconfig:option:`CONFIG_SHELL_GETOPT` set to ``y``.
+Thread safe getopt functionality is enabled by defining `__need_getopt_newlib`
```

Comment:
Would it be possible to locally define `__need_getopt_newlib` for newlib (and picolibc) in Zephyr so that the declaration is always visible?